client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -112,6 +112,7 @@ public class TrinoConnection
     private final AtomicReference<String> transactionId = new AtomicReference<>();
     private final Call.Factory httpCallFactory;
     private final Set<TrinoStatement> statements = newSetFromMap(new ConcurrentHashMap<>());
+    private Boolean useLegacyPreparedStatements = true;
```

Comment:
Since this functionality is simple and there's almost 0 scope for "regression" I'm ok with defaulting to use the old mechanism.

In general however we should try to enable new functionality by default to make sure we catch any issues which are present as early as possible instead of waiting for the day when we turn on the feature by default.

Reply:
Also in this class we can use just plain `boolean` instead of `Boolean` since we don't need a nullable one.

Reply:
Updated